### PR TITLE
Cal.com

### PR DESCRIPTION
**Pull request by eluce2**
## Description

Cal.com is an open-source alternative to Calendly. This extension lets you quickly copy your meeting links from Cal.com to your clipboard

## Screencast

https://share.cleanshot.com/8LrUfl

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by PeerRich** [comment]
thank you so much!

**Comment by agustif** [comment]
Link is private but https://github.com/calcom/api/pull/158

we're going to have a /me endpoint on API soon, which will return a user object with username alongside it.

It should be easy now to not have to ask for slug/username to the user?

File: extensions/cal-com-share-meeting-links/src/services/cal.com.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  teamId: null;
+  eventName: null;
+  timeZone: null;
+  periodType: string;
+  periodStartDate: string;
+  periodEndDate: string;
+  periodDays: null;
+  periodCountCalendarDays: false;
+  requiresConfirmation: false;
+  recurringEvent: null;
```
Comment (pernielsentikaer) [inline extensions/cal-com-share-meeting-links/src/services/cal.com.ts:57]:
```suggestion
  recurringEvent: null | recurringEvent;
```

File: extensions/cal-com-share-meeting-links/src/services/cal.com.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  schedulingType: null;
+  scheduleId: null;
+  price: number;
+  currency: string;
+  slotInterval: null;
+  metadata: object;
+  successRedirectUrl: null;
+}
+
+interface CalEventTypeResp {
```
Comment (pernielsentikaer) [inline extensions/cal-com-share-meeting-links/src/services/cal.com.ts:73]:
```suggestion
interface recurringEvent {
  freq: number;
  count: number;
  interval: number;
}

interface CalEventTypeResp {
```

File: extensions/cal-com-share-meeting-links/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                content={new URL(`${username}/${item.slug}`, "https://cal.com").toString()}
+                icon={Icon.Link}
+              />
+            </ActionPanel>
+          }
+        />
+      ))}
+    </List>
+  );
+}
```
Comment (pernielsentikaer) [inline extensions/cal-com-share-meeting-links/src/index.tsx:47]:
```suggestion
}

function Accessories(item: CalEventType) {
  let accessories = [];
  if (item.hidden) {
    accessories.push({ icon: Icon.EyeDisabled, text: "Hidden" });
  }

  if (item.recurringEvent) {
    accessories.push({ icon: Icon.Repeat, text: `Repeats up to ${item.recurringEvent.count} times` });
  }

  if (item.requiresConfirmation) {
    accessories.push({ icon: Icon.QuestionMarkCircle, text: "Requires confirmation" });
  }

  accessories.push({ icon: Icon.Clock, text: `${item.length} min` });

  return accessories;
}
```
Added multiple accessories, feel free to deny or polish 🙂
image

File: extensions/cal-com-share-meeting-links/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      primaryAction: { onAction: openCommandPreferences, title: "Open Preferences" },
+    });
+  }
+
+  return (
+    <List isLoading={isLoading}>
+      {items?.map((item) => (
+        <List.Item
+          key={item.id}
+          title={item.title}
```
Comment (pernielsentikaer) [inline extensions/cal-com-share-meeting-links/src/index.tsx:33]:
```suggestion
          title={item.title}
          accessories={Accessories(item)}
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @eluce2 👋 

Thanks for another awesome extension, I added a few things, please be sure to check it out 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by eluce2** [comment]
Thanks @pernielsentikaer! I like those changes, and I also added support for the freshly-released /me endpoint so we don't have to ask the user for their username as part of setup.

**Review by pernielsentikaer — APPROVED** [review]
Let's get the first version out @eluce2 🔥 Thanks for making this 🙂

**Pull request merged**